test/e2e/initial_scale_test.go

```diff
@@ -72,21 +72,49 @@ func TestInitScalePositive(t *testing.T) {
 
 	t.Log("Creating a new Service with initialScale 3 and verifying that pods are created")
 	createAndVerifyInitialScaleService(t, clients, names, 3)
+	t.Log("Updating the Service with a new initial scale")
+	patchAndVerifyInitialScaleService(t, clients, names, 0)
 }
 
 func createAndVerifyInitialScaleService(t *testing.T, clients *test.Clients, names test.ResourceNames, wantPods int) {
-	t.Log("Creating a new Service.", "service", names.Service)
+	t.Log("Creating a new Service", "service", names.Service)
 	_, err := v1test.CreateService(t, clients, names,
 		v1testing.WithConfigAnnotations(map[string]string{
 			autoscaling.InitialScaleAnnotationKey: strconv.Itoa(wantPods),
 		}))
 	if err != nil {
 		t.Fatal("Failed creating initial service:", err)
 	}
+	verifyInitialScaleService(t, clients, names, wantPods)
+}
+
+func patchAndVerifyInitialScaleService(t *testing.T, clients *test.Clients, names test.ResourceNames, wantPods int) {
+	t.Log("Updating Service", "service", names.Service)
+	service, err := clients.ServingClient.Services.Get(names.Service, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Error fetching Service %s: %v", names.Service, err)
+	}
+	_, err = v1test.PatchService(t, clients, service,
+		v1testing.WithConfigAnnotations(map[string]string{
+			autoscaling.InitialScaleAnnotationKey: strconv.Itoa(wantPods),
+		}))
+	if err != nil {
+		t.Fatal("Failed updating service:", err)
+	}
+	verifyInitialScaleService(t, clients, names, wantPods)
+}
 
+func verifyInitialScaleService(t *testing.T, clients *test.Clients, names test.ResourceNames, wantPods int) {
 	t.Logf("Waiting for Service %q to transition to Ready with %d number of pods.", names.Service, wantPods)
-	selector := fmt.Sprintf("%s=%s", serving.ConfigurationLabelKey, names.Service)
+	var gotPods int
 	if err := v1test.WaitForServiceState(clients.ServingClient, names.Service, func(s *v1.Service) (b bool, e error) {
+		if s.Generation != s.Status.ObservedGeneration || !s.Status.IsReady() {
+			return false, nil
+		}
+		if s.Status.LatestCreatedRevisionName == "" {
+			return false, fmt.Errorf("lastCreatedRevision is not present in Service status: %v", s)
+		}
+		selector := fmt.Sprintf("%s=%s", serving.RevisionLabelKey, s.Status.LatestCreatedRevisionName)
```

Comment:
could we use `k8s.io/apimachinery/pkg/labels` to avoid the string interpolation here?

Reply:
though I just noticed this is a test so not so important, oops :)